From 0c5d59e4f1b8a38cb63d9b2f19e49c14954fda31 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Thu, 22 Dec 2016 12:22:42 -0800
Subject: [PATCH 2121/2848] HADOOP-13911. Remove TRUSTSTORE_PASSWORD related
 scripts from KMS. Contributed by John Zhuge.

(cherry picked from commit 30f85d7a88a110637757cf7a1f4cdc9ed40f59fb)

Conflicts:
	hadoop-common-project/hadoop-kms/src/main/conf/kms-env.sh
	hadoop-common-project/hadoop-kms/src/main/libexec/kms-config.sh
	hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh

Change-Id: Ifa15aa2ce751e0bf68424cf0279f7f5c0358152b
---
 .../hadoop-kms/src/main/sbin/kms.sh                |   10 +++-------
 .../hadoop-kms/src/main/tomcat/ssl-server.xml.conf |    1 -
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh b/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh
index f5b9d0a..d7489a7 100644
--- a/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh
+++ b/hadoop-common-project/hadoop-kms/src/main/sbin/kms.sh
@@ -53,10 +53,7 @@ fi
 # it is used in Tomcat's server.xml configuration file
 #
 
-# Mask the trustStorePassword
-KMS_SSL_TRUSTSTORE_PASS=`echo $CATALINA_OPTS | grep -o 'trustStorePassword=[^ ]*' | awk -F'=' '{print $2}'`
-CATALINA_OPTS_DISP=`echo ${CATALINA_OPTS} | sed -e 's/trustStorePassword=[^ ]*/trustStorePassword=***/'`
-print "Using   CATALINA_OPTS:       ${CATALINA_OPTS_DISP}"
+print "Using   CATALINA_OPTS:       ${CATALINA_OPTS}"
 
 catalina_opts="-Dkms.home.dir=${KMS_HOME}";
 catalina_opts="${catalina_opts} -Dkms.config.dir=${KMS_CONFIG}";
@@ -81,14 +78,13 @@ if [ "${1}" = "stop" ]; then
 fi
 
 # If ssl, the populate the passwords into ssl-server.xml before starting tomcat
-if [ ! "${KMS_SSL_KEYSTORE_PASS}" = "" ] || [ ! "${KMS_SSL_TRUSTSTORE_PASS}" = "" ]; then
+if [ ! "${KMS_SSL_KEYSTORE_PASS}" = "" ]; then
   # Set a KEYSTORE_PASS if not already set
   KMS_SSL_KEYSTORE_PASS=${KMS_SSL_KEYSTORE_PASS:-password}
   KMS_SSL_KEYSTORE_PASS_ESCAPED=$(hadoop_escape "$KMS_SSL_KEYSTORE_PASS")
-  KMS_SSL_TRUSTSTORE_PASS_ESCAPED=$(hadoop_escape "$KMS_SSL_TRUSTSTORE_PASS")
   cat ${CATALINA_BASE}/conf/ssl-server.xml.conf \
     | sed 's/"_kms_ssl_keystore_pass_"/'"\"${KMS_SSL_KEYSTORE_PASS_ESCAPED}\""'/g' \
-    | sed 's/"_kms_ssl_truststore_pass_"/'"\"${KMS_SSL_TRUSTSTORE_PASS_ESCAPED}\""'/g' > ${CATALINA_BASE}/conf/ssl-server.xml
+    > ${CATALINA_BASE}/conf/ssl-server.xml
   cp ${CATALINA_BASE}/conf/ssl-server.xml ${CATALINA_BASE}/conf/server.xml
 else
   cp ${CATALINA_BASE}/conf/server.xml.conf ${CATALINA_BASE}/conf/server.xml
diff --git a/hadoop-common-project/hadoop-kms/src/main/tomcat/ssl-server.xml.conf b/hadoop-common-project/hadoop-kms/src/main/tomcat/ssl-server.xml.conf
index 01b429c..272542a 100644
--- a/hadoop-common-project/hadoop-kms/src/main/tomcat/ssl-server.xml.conf
+++ b/hadoop-common-project/hadoop-kms/src/main/tomcat/ssl-server.xml.conf
@@ -72,7 +72,6 @@
                maxThreads="${kms.max.threads}" scheme="https" secure="true"
                maxHttpHeaderSize="${kms.max.http.header.size}"
                clientAuth="false" sslEnabledProtocols="TLSv1,TLSv1.1,TLSv1.2,SSLv2Hello"
-               truststorePass="_kms_ssl_truststore_pass_"
                keystoreFile="${kms.ssl.keystore.file}"
                keystorePass="_kms_ssl_keystore_pass_"/>
 
-- 
1.7.9.5

